lib/request.js

```diff
@@ -0,0 +1,34 @@
+module.exports = request
+
+var Promise = require('./utils/promise')
+
+var internals = module.exports.internals = {}
+internals.request = require('./utils/request')
+
+function request (state, options) {
+  if (!options) {
+    return Promise.reject(new Error('hoodie.request: URL or options argument must be defined'))
+  }
+
+  var requestOptions = {
+    url: options.url || options
+  }
+
+  if ((/^\/(?!\/)/).test(requestOptions.url)) {
+    requestOptions.url = state.url + requestOptions.url
+  }
+
+  if (options.data) {
+    requestOptions.body = options.data.toString()
```

Comment:
Feel free to use JSON.stringify directly, with PouchDB we have a stricter browser requirements already :) 

(Even though we'd need to formally document them at some point)